source/github-helpers/github-url.ts

```diff
@@ -37,17 +37,18 @@ export default class GitHubURL {
 		ambiguousReference: string[]
 	): {branch: string; filePath: string} {
 		const branch = ambiguousReference[0];
+		const currentBranch = getCurrentCommittish();
 		// History pages might use search parameters
+		const branchFromSearch = String(this.searchParams.getAll('branch'));
 		const filePathFromSearch = this.searchParams.getAll('path[]').join('/');
 		if (filePathFromSearch) {
+			this.searchParams.delete('branch');
 			this.searchParams.delete('path[]');
-			return {branch, filePath: filePathFromSearch};
+			return {branch: branchFromSearch ?? currentBranch ?? branch, filePath: filePathFromSearch};
```

Comment:
When would this link have a file path but not a branch? Steps to reproduce please

Reply:
https://github.com/yakov116/refined-github/commits/f23b687b3b89aa95a76193722cdfeff740646670?after=f23b687b3b89aa95a76193722cdfeff740646670+34&path%5B%5D=source&path%5B%5D=features&path%5B%5D=release-download-count.tsx

Reply:
Steps to reproduce. That's just a link that anyone can write manually but maybe doesn't exist

Reply:
That is the url we were testing until now. The branch in the path is a new thing.

Reply:
That URL does not have a `branch` URL parameter

Reply:
https://github.com/sindresorhus/refined-github/commits/Slash/slash?after=ee8b6422d2ddd1199f557594d03e7ee850e843f0+34&branch=Slash%2Fslash&path%5B%5D=webpack.config.ts

Reply:
> That URL does not have a `branch` URL parameter (`branchFromSearch`). Where did you see that?

Oh my I just re read you comment 🤦 

That url is the one we are testing until now. GitHub recently starting adding the branch to the search

Reply:
Once again, I need steps to get that URL. I want a starting URL that isn’t what you’re pasting here. I can also make up a URL and send a comment, that doesn’t mean it exists anywhere else.

Reply:
This is what I wanted:

1. Visit https://github.com/sindresorhus/refined-github/commits/Slash/slash/webpack.config.ts
2. Click "Older"

Reply:
Anyway, this confirms that there's no way that `filePathFromSearch` is truey while `branchFromSearch` isn't, so it should probably be
```suggestion
			return {branch: branchFromSearch, filePath: filePathFromSearch};
```

Reply:
GHE. The branch part is a new thing (at least it was not there when we made GitHubURL

Reply:
Did the `pathname` look exactly like now too? Because we can easily extract the slashed branch from there instead.

Reply:
Yes it did.